watertap/property_models/tests/test_multicomp_aq_sol_prop_pack.py

```diff
@@ -2485,3 +2507,625 @@ def test_automatic_charge_mw_population():
+            material_flow_basis=MaterialFlowBasis.mass,
+            diffusivity_data={
+                ("Liq", "Ca_2+"): 7.92e-10,
+                ("Liq", "SO4_2-"): 1.06e-09,
+                ("Liq", "Na_+"): 1.33e-09,
+                ("Liq", "Cl_-"): 2.03e-09,
+                ("Liq", "Mg_2+"): 7.06e-10,
+            },
+            mw_data={
+                "H2O": 0.018,
+                "Na_+": 0.023,
+                "Ca_2+": 0.04,
+                "Mg_2+": 0.024,
+                "Cl_-": 0.035,
+                "SO4_2-": 0.096,
+            },
+            stokes_radius_data={
+                "Na_+": 1.84e-10,
+                "Ca_2+": 3.09e-10,
+                "Mg_2+": 3.47e-10,
+                "Cl_-": 1.21e-10,
+                "SO4_2-": 2.3e-10,
+            },
+            charge={"Na_+": 1, "Ca_2+": 2, "Mg_2+": 2, "Cl_-": -1, "SO4_2-": -2},
+            elec_mobility_calculation=ElectricalMobilityCalculation.EinsteinRelation,
+            density_calculation=DensityCalculation.seawater,
+            activity_coefficient_model=ActivityCoefficientModel.davies,
+        )
+
+        m.fs.stream = stream = m.fs.properties.build_state_block(
+            [0], defined_state=True
+        )
+
+        mass_flow_in = 1 * pyunits.kg / pyunits.s
+        feed_mass_frac = {
+            "Na_+": 11122e-6,
+            "Ca_2+": 382e-6,
+            "Mg_2+": 1394e-6,
+            "SO4_2-": 2136e-6,
+            "Cl_-": 20300e-6,
+        }
+        for ion, x in feed_mass_frac.items():
+            mass_comp_flow = x * pyunits.kg / pyunits.kg * mass_flow_in
+
+            stream[0].flow_mass_phase_comp["Liq", ion].fix(mass_comp_flow)
+
+        H2O_mass_frac = 1 - sum(x for x in feed_mass_frac.values())
+
+        stream[0].flow_mass_phase_comp["Liq", "H2O"].fix(H2O_mass_frac)
+        stream[0].temperature.fix(298.15)
+        stream[0].pressure.fix(101325)
+
+        stream[0].assert_electroneutrality(
+            defined_state=True, tol=1e-2, adjust_by_ion="Cl_-"
+        )
+
+        metadata = m.fs.properties.get_metadata().properties
+        for v in metadata.list_supported_properties():
+            getattr(stream[0], v.name)
+            assert stream[0].is_property_constructed(v.name)
+
+        initializer = stream.default_initializer()
+        initializer.initialize(stream)
+        assert initializer.summary[stream]["status"] == InitializationStatus.Ok
+
+        scaler = stream.default_scaler()
+
+        set_scaling_factor(stream[0].flow_mass_phase_comp["Liq", "H2O"], 1)
+        set_scaling_factor(stream[0].flow_mass_phase_comp["Liq", "Ca_2+"], 1e4)
+        set_scaling_factor(stream[0].flow_mass_phase_comp["Liq", "SO4_2-"], 1e3)
+        set_scaling_factor(stream[0].flow_mass_phase_comp["Liq", "Na_+"], 1e2)
+        set_scaling_factor(stream[0].flow_mass_phase_comp["Liq", "Cl_-"], 1e2)
+        set_scaling_factor(stream[0].flow_mass_phase_comp["Liq", "Mg_2+"], 1e3)
+
+        scaler.scale_model(stream[0])
```

Comment:
For that you do not want to look at this: https://github.com/IDAES/idaes-compatibility/tree/main/src/idaes_compatibility/robustness

Rather than look at some numerical characteristic of the model, you want to look at how well the solver performed at solving it. Additionally, it cannot be a single point test, but needs to cover a range of conditions.

Reply:
That would be perfect! I didn't know about this IPOPTConvergenceAnalysis class. Didn't explore its full functionality yet, but if I am understanding correctly, then I would say we should test a range of conditions to check how the scaler affects convergence.

I would even say that we can think about a way to apply that to probing how "good" an initializer is as well, granted we'd have to follow up the initialization with a solve if it wasn't baked in already.

Reply:
I will note the plan is/was to move IPOPTConvergenceAnalysis to sue the parameter sweep tool once it is ready (I tried to make it mostly compatible).